scripts/core/exp-command-buffer.yml

```
@@ -54,7 +87,120 @@ type: struct
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Descriptor type for updating a kernel command pointer argument."
+base: $x_base_desc_t
+name: $x_exp_command_buffer_update_pointer_arg_desc_t
+members:
+    - type: uint32_t
+      name: argIndex
+      desc: "[in] Argument index."
+    - type: "const ur_kernel_arg_pointer_properties_t *"
+      name: pProperties
+      desc: "[in][optinal] Pointer to USM pointer properties."
+    - type: "const void *"
+      name: pArgValue
+      desc: "[in][optional] USM pointer to memory location holding the argument value."
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Descriptor type for updating a kernel command value argument."
+base: $x_base_desc_t
+name: $x_exp_command_buffer_update_value_arg_desc_t
+members:
+    - type: uint32_t
+      name: argIndex
+      desc: "[in] Argument index."
+    - type: uint32_t
+      name: argSize
+      desc: "[in] Argument size."
+    - type: "const ur_kernel_arg_value_properties_t *"
+      name: pProperties
+      desc: "[in][optinal] Pointer to memory object properties."
+    - type: "const void *"
+      name: pArgValue
+      desc: "[in][optional] Argument value representing kernel arg type."
+
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Descriptor type for updating kernel command execution info."
+base: $x_base_desc_t
+name: $x_exp_command_buffer_update_exec_info_desc_t
+members:
+    - type: ur_kernel_exec_info_t
+      name: propName
+      desc: "[in] Name of execution attribute."
+    - type: size_t
+      name: propSize
+      desc: "[in] Size of execution attribute."
+    - type: "const ur_kernel_exec_info_properties_t *"
+      name: pProperties
+      desc: "[in][optional] Pointer to execution info properties."
+    - type: "const void *"
+      name: pPropValue
+      desc: "[in] Pointer to memory location holding the property value."
+
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Descriptor type for updating a kernel launch command."
+base: $x_base_desc_t
+name: $x_exp_command_buffer_update_kernel_launch_desc_t
+members:
+    - type: uint32_t
+      name: numMemobjArgs
+      desc: "[in] Length of pArgMemobjList."
+    - type: uint32_t
+      name: numPointerArgs
+      desc: "[in] Length of pArgPointerList."
+    - type: uint32_t
+      name: numValueArgs
+      desc: "[in] Length of pArgValueList."
+    - type: uint32_t
+      name: numExecInfos
+      desc: "[in] Length of pExecInfoList."
+    - type: uint32_t
+      name: workDim
+      desc: "[in][optional] Number of work dimensions in the kernel ND-range, from 1-3."
+    - type: "const $x_exp_command_buffer_update_memobj_arg_desc_t*"
+      name: pArgMemobjList
+      desc: "[in][optional] An array describing the new kernel mem obj arguments for the command."
```

can you add range to all of these so the scripts know they're arrays, e.g.
```suggestion
      desc: "[in][optional][range(0, numMemobjArgs)] An array describing the new kernel mem obj arguments for the command."
```